handbook/engineering/hiring/software-engineer-frontend.md

```diff
@@ -46,18 +53,30 @@ Learn more about what it is like to work at Sourcegraph by reading [our handbook
 ## Interview process
 
 1. You [apply here](https://jobs.lever.co/sourcegraph/b2f9a8b0-cc06-4629-81a0-0f2fa64271c7/apply).
-1. We set up a 30-minute call to learn more about what you are looking for, tell you about Sourcegraph, and answer any questions that you have.
-1. You complete a 3-hour [exercise project](software-engineer-coding-exercise.md#frontend-coding-exercise) that we designed to test the technical qualifications listed above.
-1. We review your submission and schedule a **1 hour follow-up call** to discuss it.
-1. We schedule additional interviews across multiple days.
-   - 1h **Technical experience:** We ask you about your past work and accomplishments in depth, how you worked with others, decisions you made, and what you'd do differently today.
-   - 1h **Team collaboration:** We ask you about how you work and communicate in a team setting, and how you handle tricky situations.
-   - 30m **VP Engineering**
-   - 30m **CTO**
-   - 30m **CEO**
+1. The **Hiring Manager** screens your application for suitability for the role by looking at:
+   - The answers you provide to the application questions.
+   - The skills you have obtained in your career.
+   - Your past work experience. In the absence of professional work experience we would look at any open-source projects you have created or contribute to.
+   - The [Nice-to-haves](#nice-to-haves) we listed above.
+1. A **Recruiter** sets up a 30 minute call to learn more about what you are looking for, tell you about Sourcegraph, and answer any questions that you have.
+1. We set up a 1 hour interview with the **Hiring Manager**, who will tell you more about the role in the team. They will ask you about your past work and accomplishments and assess your sutiability for the role.
+1. You complete a 3 hour [coding excercise](software-engineer-coding-exercise.md#frontend-coding-exercise) that we designed to test your technical ability and the responsibilities of the role as [listed above](#your-responsibilities).
+   - _Note: while we do not disqualify candidates who do not have React and TypeScript experience from taking the excercise, you will be required to use these for the excercise._
+1. An **Engineer** and a **Designer** will review and grade your submission.
+1. We schedule a **1 hour follow-up call** to discuss your submission and ask any questions we might have to clarify our understanding of your technical capability and sutiability for the role.
+1. We schedule the following additional interviews, in no particular order, across multiple days:
+   - 1 hour **Technical experience:** We ask you about your past work and accomplishments in depth, how you worked with others, decisions you made, and what you'd do differently today.
```

Comment:
Do we still need this or is this duplicative of the HM interview? I want to ensure that candidates aren't telling the same story over and over again.

Reply:
@nicksnyder I was wondering about this myself and agree it could lead to too much repetition. I'd say it's not needed based on the HM interview that proceeds it and other team members or frontend engineers can join the team collaboration interview rather.

Reply:
I'd be worried about removing this interview. It has a great documented script, has a trained interviewer pool of engineers, is very well calibrated and the only other somewhat technical assessment other than the coding exercise (which is also still new and not well calibrated). I would feel less confident in making offers to candidates if this interview didn't exist.

Reply:
@felixfbecker I don't think the idea is to remove the value this interview brought to the overall decision making, but rather that it falls on the hiring manager to cover this in their interview with the candidate. 

I found myself repeating a lot of information during my interviews at Sourcegraph, and previous roles for that matter, and it's always bugged me that companies with many interview steps don't make sure the context from previous interviews are shared so that candidates don't have to repeat themselves.

Reply:
Imo the Team Collaboration interview is not equivalent to Technical Experience:
- [Team Collaboration](https://github.com/sourcegraph/interviews/blob/master/engineering/team-collaboration.md)
- [Technical Experience](https://github.com/sourcegraph/interviews/blob/master/engineering/technical-experience.md)

In Team Collaboration there is no test for communicating with a _technical_ audience, continuous learning, mentorship nor interest/passion. There is maybe some overlap in the communication part, but the important difference here was that one is communication with a technical audience and the other with a non-technical audience (and that's why the interview panels are engineers for Technical Experience and non-engineers for Team Collaboration).

It _may_ be the case that a situation a candidate went through is a suitable example for a question in either interview but they would still be talking about separate aspects of that situation and ideally they can come up with separate, targeted examples (i.e. it's on them if they only have a single story and it would probably not work in their favor, unless that story is faceted enough to serve as an example for all the different questions).

If you care strongly about removing this and want to keep this change, @jeanduplessis what change do you propose to the Team Collaboration interview panel and what is the script for the HM call (so we make sure we're not removing any of these assessments from the overall interview process)?

Reply:
One side-effect of this would also be that the candidate would meet 2 less people of the team in the interview process (e.g. only one non-technical team member).

Reply:
> I found myself repeating a lot of information during my interviews at Sourcegraph

This is a valuable data point and I'd be curious to here more about this from both your and interviewer's perspectives (as I wasn't part of your interview panel).

> it's always bugged me that companies with many interview steps don't make sure the context from previous interviews are shared so that candidates don't have to repeat themselves.

As an interviewer at Sourcegraph, I have also previously wished we would share more context between interviews so we could know what to dig in more on, but what I heard was that it's intentional that we don't get to see other's feedback until the debrief to not introduce bias and that interview modules should be stand-alone and be able to each independently test a subset of the overall requirements. Curious if you agree with that?

Reply:
> One side-effect of this would also be that the candidate would meet 2 less people of the team in the interview process (e.g. only one non-technical team member).

I would imagine we could incorporate the technical team members as part of the team collaboration session?

> This is a valuable data point and I'd be curious to here more about this from both your and interviewer's perspectives (as I wasn't part of your interview panel).

As the candidate, I found I had to share my work background and experience quite a bit. I can't exactly recall if it was any specific question that triggered that, but I do recall it happening frequently enough that I made a mental note of wanting to address it.

In terms of sharing context, I think what we want to avoid is sharing opinions about a candidate if there are other interviews still taking place. However, I see this as being different from a summary of a person's background and experience (i.e. more colorful version that their CV) being passed on as context for future interviewers. In fact, I would expect those interviewers to review the context and possibly ask follow-up questions of the candidate based on it. What I like to do as well is highlight areas I want another interviewer to dig into further, either because I am uncertain about something, or I didn't have time to get deeper into it.

Reply:
General principles:
- We need to ensure our hiring process captures all the signal that we care about.
- We need to ensure candidates aren't repeating themselves.

We generally accomplish both of these things by writing down the questions we ask in each interview so we can audit them to ensure nothing is duplicated.

In this diff we added a 1hr HM interview, and deleted the 1h technical experience interview. I think the current gap in the documentation is that we don't have guidance for the 1hr HM interview. Felix has concerns that we will no longer be getting important signal that we used to get from the technical experience interview.

@jeanduplessis Can you please open a new PR that either (1) documents the HM is doing the technical experience interview (to a first approximation) or (2) lists the questions that the HM is going to ask so we can audit it to ensure we get the right signal.


****


The second point is sharing candidate background across interviews. I agree this would be valuable but there are some challenges we would need to figure out.

1. We don't want to share the interviewer's judgement about the background, just the background itself.
2. When interviews are back-to-back, there is no time for interviewers to (A) write down the relevant info to share and (B) review feedback from previous interviews. (A) could be solved with transcription software like otter.ai. (B) could only be solved by scheduling gaps between interviews.

Another option is to move more of our interview process to "answer these written questions" (which anyone on the team can review) and then we have a live discussion.